### PR TITLE
Switch GCR -> Artifact-Registry

### PR DESCRIPTION
**Pull request by ccwienk**
GCR has been deprecated [0] in favour of Artifact-Registry.

Thus, change push-targets for OCI-Images:

- europe-docker.pkg.dev/gardener-project/snapshots for snapshots
- europe-docker.pkg.dev/gardener-project/releases for releases

To allow users for smooth transitioning, keep also publishing to GCR.

[0]
https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr

**Release note**:

```breaking operator
Change OCI Image Registry from GCR (`eu.gcr.io/gardener-project`) to Artifact-Registry (`europe-docker.pkg.dev/gardener-project/releases`). Users should update their references.

```

### CODE REVIEW

File: .ci/pipeline_definitions [hunk 1]
```diff
@@ -18,8 +19,9 @@ gardener-extension-networking-cilium:
       publish:
         dockerimages:
           gardener-extension-networking-cilium:
-            registry: 'gcr-readwrite'
-            image: 'eu.gcr.io/gardener-project/gardener/extensions/networking-cilium'
+            image: 'europe-docker.pkg.dev/gardener-project/snapshots/extensions/networking-cilium'
+            extra_push_targets: # may be dropped after all users updated to new registry
+              - eu.gcr.io/gardener-project/gardener/extensions/networking-cilium
```
[the remark this answers is not in the document]
Comment (ccwienk) [reply .ci/pipeline_definitions:24]: @ScheererJ : @oliver-goetz proposed to publish to two locations to make migration easier. If you prefer to migrate without such an intermediate step, I can remove the `extra_push_targets` from the pull-request (fine w/ me)

File: .ci/pipeline_definitions [hunk 2]
```diff
@@ -60,6 +63,8 @@ gardener-extension-networking-cilium:
       traits:
         version:
           preprocess: 'finalize'
+        component_descriptor:
+          ocm_repository: europe-docker.pkg.dev/gardener-project/releases
```
[the remark this answers is not in the document]
Comment (ccwienk) [reply .ci/pipeline_definitions:67]: the `ocm_repository` defines the push-target, whereas the mapping defines optional additional repositories from which to lookup component-descriptors.

File: test/templates/network-connectivity-test.yaml.tpl [hunk 3]
```diff
@@ -13,7 +13,7 @@ spec:
         app: networking-test
     spec:
       containers:
-      - image: eu.gcr.io/gardener-project/gardener/cilium-cli:1.2.0
+      - image: europe-docker.pkg.dev/gardener-project/releases/gardener/cilium-cli:1.2.0
```
[the remark this answers is not in the document]
Comment (ccwienk) [reply test/templates/network-connectivity-test.yaml.tpl:16]: yes

File: .ci/pipeline_definitions (hunk 2 above)
[the remark this answers is not in the document]
Comment (ScheererJ) [reply .ci/pipeline_definitions:67]: Why is the additional mapping needed then?

File: .ci/pipeline_definitions (hunk 1 above)
[the remark this answers is not in the document]
Comment (ScheererJ) [reply .ci/pipeline_definitions:24]: @DockToFuture / @axel7born : WDYT?

File: .ci/pipeline_definitions [hunk 4]
```diff
@@ -18,8 +19,9 @@ gardener-extension-networking-cilium:
       publish:
         dockerimages:
           gardener-extension-networking-cilium:
-            registry: 'gcr-readwrite'
```
[the remark this answers is not in the document]
Comment (ccwienk) [reply .ci/pipeline_definitions:21]: @DockToFuture : in a way, yes. we calculate the appropriate credentials based on metadata we keep together w/ the credentials. it is of course still supported to explicitly overwrite the automatic credentials-selection. however, I think it is better to not do so and thus not hard-code cfg names in our pipeline-definitions

File: hack/test-e2e-provider-local.sh [hunk 5]
```diff
@@ -30,7 +30,7 @@ cd $repo_root
 
 version=$(git rev-parse HEAD)
 make docker-images
-docker tag eu.gcr.io/gardener-project/gardener/extensions/networking-cilium:latest networking-cilium-local:$version
+docker tag europe-docker.pkg.dev/gardener-project/releases/extensions/networking-cilium:latest networking-cilium-local:$version
```
Comment (DockToFuture) [inline hack/test-e2e-provider-local.sh:33]: Wouldn't it make more sense to use the public image here?

File: Makefile [hunk 6]
```diff
@@ -15,7 +15,7 @@
 EXTENSION_PREFIX            := gardener-extension
 NAME                        := networking-cilium
 ADMISSION_NAME              := admission-cilium
-REGISTRY                    := eu.gcr.io/gardener-project/gardener
+REGISTRY                    := europe-docker.pkg.dev/gardener-project/releases
```
Comment (ScheererJ) [inline Makefile:18]: Why does this differ from how we do it for `calico`, i.e. why not use `europe-docker.pkg.dev/gardener-project/public/gardener`?

**Review by ScheererJ — CHANGES_REQUESTED** [review]

File: .ci/pipeline_definitions (hunk 2 above)
[the remark this answers is not in the document]
Comment (ccwienk) [reply .ci/pipeline_definitions:67]: sure, I will change it like that. I agree explicitness and thus "readability" seems more important than a negligible deduplication

File: Makefile (hunk 6 above)
Comment (ccwienk) [reply Makefile]: changed to `/public/gardener`, as suggested. btw: both previously, and after this change, `registry` is not an adequate term according to naming according to OCI.

[the remark this answers is not in the document]
Comment (ccwienk) [reply Makefile]: changed it

File: .ci/pipeline_definitions (hunk 1 above)
[the remark this answers is not in the document]
Comment (ccwienk) [reply .ci/pipeline_definitions:22]: did you make up your mind w.r.t. "extra_push_targets"?

File: hack/test-e2e-provider-local.sh (hunk 5 above)
Comment (ccwienk) [reply hack/test-e2e-provider-local.sh]: done

File: .ci/pipeline_definitions (hunk 1 above)
[the remark this answers is not in the document]
Comment (ScheererJ) [reply .ci/pipeline_definitions:22]: We did not reach a clear conclusion. Let's keep it as is for now.

**Review by ScheererJ — APPROVED** [review]
Thanks for the changes.

/lgtm

**Pull request merged**